scripts/remove-idle-users.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python
+from __future__ import print_function
+import argparse
+import sys
+import json
+import urllib
+import requests
+import re
+
+## debug request
+import httplib as http_client
+http_client.HTTPConnection.debuglevel = 1
+
+def get_room_id(homeserver, alias, token):
+    res = requests.get(homeserver + "/_matrix/client/r0/directory/room/" + urllib.quote(alias) + "?access_token=" + token).json()
+    return res.get("room_id", None)
+
+def get_last_active_ago(homeserver, user_id, token):
+    res = requests.get(homeserver + "/_matrix/client/r0/presence/" + urllib.quote(user_id) + "/status?access_token=" + token).json()
+    return res.get("last_active_ago", None)
+
+def is_idle(homeserver, user_id, token, activity_threshold_ms):
+    return get_last_active_ago(homeserver, user_id, token) > activity_threshold_ms
+
+def get_idle_users(homeserver, room_id, token, since):
+    res = requests.get(homeserver + "/_matrix/client/r0/rooms/" + urllib.quote(room_id) + "/members?access_token=" + token)
+    user_ids = [event["state_key"] for event in res.json().get("chunk", None)]
+
+    activity_threshold_ms = since * 24 * 60 * 60 * 1000
+
+    return [user_id for user_id in user_ids if is_idle(homeserver, user_id, token, activity_threshold_ms)]
+
+def kick_idlers(homeserver, homeserver_domain, room_id, token, since, user_template=None):
+    reason = "Being idle for >%s days" % since
+
+    user_ids = get_idle_users(homeserver, room_id, token, since)
+    print("Kicking %s idle users from %s" % (len(user_ids), room_id))
+    for user_id in user_ids:
+        # Ignore unclaimed users, if user_template is specified
+        if user_template and not claims_user_id(user_id, user_template, homeserver_domain):
+            continue
+        res = requests.post(
+            homeserver + "/_matrix/client/r0/rooms/" + urllib.quote(room_id) + "/kick?access_token=" + token,
+            data = json.dumps({
+                "reason": reason,
+                "user_id": user_id
+            })
+        )
+        res.raise_for_status()
```

Comment:
Don't raise for status here. It'll be annoying as all hell if it spends the time working out who to kick then dies on the first failure. I would add the failures to a list and then whine at the end saying "I couldn't kick these guys because $RESPONSE_JSON".